### ISSUE TITLE
"error":"popup_closed_by_user"

### ISSUE THREAD

**Issue by josemojica023** [abacritt/angularx-social-login#388]
![Screen Shot 2021-05-04 at 1 56 09 PM](https://user-images.githubusercontent.com/68880820/116965478-82ad9d80-ace0-11eb-8412-46899ee02a1e.png)
This will happen when a uses close the pop up modal for choosing their accounts. Need to handle this error.

**Comment by gedkul** [comment]
Create new "OAuth 2.0 Client IDs" again and specify domains that  u use. like : http://localhost:4200 and if u use production specify that domain too.

**Comment by jaibatrik** [comment]
I think the `Promise` returned by `signIn` method rejects in this case, and you can handle the rejection. Can you check if that helps?

**Comment by TarasKovalenko** [comment]
@jaibatrik @josemojica023 
yep, the promise catch method will return this error and you can easily handle this action if you want:
```
this.authService.signIn(GoogleLoginProvider.PROVIDER_ID)
      .then(d => console.log(d))
      .catch(error => {
        console.log(error);
      });
```

**Comment by TLKG** [comment]
@jaibatrik @TarasKovalenko  I'm having this issue even my "locahost:4200" is added to the **Allow** list. I added `.catch` part suggested by @TarasKovalenko , it still not resolving `{error: 'popup_closed_by_user'}`
Any other solution? Thank you and appreciate it.

**Comment by jaibatrik** [comment]
I think you should handle it in the catch of Promise returned by signIn method.

**Comment by TLKG** [comment]
HTML and TS:

```
<div style=" display: flex; justify-content: center;">
        <button (click)="loginWithGoogle()" style="border: none; background-color: transparent; outline: none;"><img src="assets/img/google.png" alt="sign in"></button>
      </div>

import { SocialAuthService, GoogleLoginProvider, SocialUser } from 'angularx-social-login';

constructor
  (
      private socialAuthService: SocialAuthService
  )

loginWithGoogle(): void {
    this.socialAuthService.signIn(GoogleLoginProvider.PROVIDER_ID)
        .then(d => console.log(d))
        .catch(error => { console.log(error);});
  }
```

@jaibatrik Would you please elaborate where and how to handle the `popup close by user` if this is not implemented right? Thank you very much!

**Comment by jaibatrik** [comment]
This actually looks alright to me. If this doesn't help, reopen the issue, I'll try to check it. Not getting time these days though.

**Comment by TLKG** [comment]
@jaibatrik Appreciate it!
More details:
- It only started on F5 debug session a few weeks ago while no new/update packages, except VSC auto-updated itself 
- adding `cross-origin-opener-policy` to angular.json based on this [SO](https://stackoverflow.com/questions/48683320/google-sso-login-error-popup-closed-by-user/66290764#66290764), not work.
- Sometimes removing all Google acct from Chrome works.
- Sometimes a complete logout from Google works.
- Sometimes disable all breakpoint, then enable all works.
- Sometimes add new breakpoint works.
- Sometimes delete node_modules folder and `npm install` works.
- Clear Chrome cache never works.
- Adding `https://localhost:4200`, `http://localhost:4200`, and allow popup for **ALL** sites at Chrome never works.
- Adding followings to `node_modules\istanbul-reports\lib\html-spa\webpack.config.js` under module.exports, output, not helping:
        devtoolModuleFilenameTemplate: '[absolute-resource-path]'
- First F5 run of a new day always get error now, regardless Chrome or Edge.
- **Standalone** Chrome and Edge **works always**. 

**Angular CLI: 11.2.6**
Node: 14.15.5
OS: win32 x64

Angular:
...
Ivy Workspace:

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1102.6 (cli-only)
@angular-devkit/core         11.2.6 (cli-only)
@angular-devkit/schematics   11.2.6 (cli-only)
@schematics/angular          11.2.6 (cli-only)
@schematics/update           0.1102.6 (cli-only)

**VSC**
Version: 1.60.2 (user setup)
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T12:00:31.514Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

**Comment by TLKG** [comment]
@jaibatrik  If ng 12 & angularx-social-login v4 never has this issue or doesn't use popup, I'll try to upgrade and test if works. There is no option for me to re-open this issue. Thank you very much again!

**Comment by TLKG** [comment]
It should be an NVM issue.

Details:

- I've been using nvm for two years and no problem.
- when this problem started I noticed Chrome loading first login page very slow, 7 sec. vs. 2 sec.
- Tried the following not helping
downgrade VSC to 2021/03 while sticking with ng11 and login3.
upgraded to ng12 and login4 for both VSC 2021/3 and current.

- After removing NVM, current VSC works with ng12 and login4.
- Only a few times I had to remove cached login acct in Chrome, `npm cache clear`, restart VSC
- So far it's been ok for over 5 days, and first login page loads fast as expected.
- meanwhile, on another machine never had VSC and nvm, ALWAYS ok with current VSC and ng12/login4.
- What two machines have in common is breakpoints at `.then()` and/or `.catch()` won't continue.

```
this.socialAuthService.signIn(GoogleLoginProvider.PROVIDER_ID)
        .then(d => console.log(d))     // breakpoint here bad
        .catch(error => { console.log(error);});    // breakpoint here bad

```

Post here for anyone might encounter the same problem. Thank you all.
